contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -1283,146 +1287,298 @@ sp_describe_undeclared_parameters_internal(PG_FUNCTION_ARGS)
-		"WHEN T2.name COLLATE sys.database_default = \'bigint\' THEN 127 "
-		"WHEN T2.name COLLATE sys.database_default = \'binary\' THEN 173 "
-		"WHEN T2.name COLLATE sys.database_default = \'bit\' THEN 104 "
-		"WHEN T2.name COLLATE sys.database_default = \'char\' THEN 175 "
-		"WHEN T2.name COLLATE sys.database_default = \'date\' THEN 40 "
-		"WHEN T2.name COLLATE sys.database_default = \'datetime\' THEN 61 "
-		"WHEN T2.name COLLATE sys.database_default = \'datetime2\' THEN 42 "
-		"WHEN T2.name COLLATE sys.database_default = \'datetimeoffset\' THEN 43 "
-		"WHEN T2.name COLLATE sys.database_default = \'decimal\' THEN 106 "
-		"WHEN T2.name COLLATE sys.database_default = \'float\' THEN 62 "
-		"WHEN T2.name COLLATE sys.database_default = \'image\' THEN 34 "
-		"WHEN T2.name COLLATE sys.database_default = \'int\' THEN 56 "
-		"WHEN T2.name COLLATE sys.database_default = \'money\' THEN 60 "
-		"WHEN T2.name COLLATE sys.database_default = \'nchar\' THEN 239 "
-		"WHEN T2.name COLLATE sys.database_default = \'ntext\' THEN 99 "
-		"WHEN T2.name COLLATE sys.database_default = \'numeric\' THEN 108 "
-		"WHEN T2.name COLLATE sys.database_default = \'nvarchar\' THEN 231 "
-		"WHEN T2.name COLLATE sys.database_default = \'real\' THEN 59 "
-		"WHEN T2.name COLLATE sys.database_default = \'smalldatetime\' THEN 58 "
-		"WHEN T2.name COLLATE sys.database_default = \'smallint\' THEN 52 "
-		"WHEN T2.name COLLATE sys.database_default = \'smallmoney\' THEN 122 "
-		"WHEN T2.name COLLATE sys.database_default = \'text\' THEN 35 "
-		"WHEN T2.name COLLATE sys.database_default = \'time\' THEN 41 "
-		"WHEN T2.name COLLATE sys.database_default = \'tinyint\' THEN 48 "
-		"WHEN T2.name COLLATE sys.database_default = \'uniqueidentifier\' THEN 36 "
-		"WHEN T2.name COLLATE sys.database_default = \'varbinary\' THEN 165 "
-		"WHEN T2.name COLLATE sys.database_default = \'varchar\' THEN 167 "
-		"WHEN T2.name COLLATE sys.database_default =  \'xml\' THEN 241 "
-		"ELSE C.system_type_id "
+		", CASE T2.name COLLATE sys.database_default "
+		"WHEN \'bigint\' THEN 127 "
+		"WHEN \'binary\' THEN 173 "
+		"WHEN \'bit\' THEN 104 "
+		"WHEN \'char\' THEN 175 "
+		"WHEN \'date\' THEN 40 "
+		"WHEN \'datetime\' THEN 61 "
+		"WHEN \'datetime2\' THEN 42 "
+		"WHEN \'datetimeoffset\' THEN 43 "
+		"WHEN \'decimal\' THEN 106 "
+		"WHEN \'float\' THEN 62 "
+		"WHEN \'image\' THEN 34 "
+		"WHEN \'int\' THEN 56 "
+		"WHEN \'money\' THEN 60 "
+		"WHEN \'nchar\' THEN 239 "
+		"WHEN \'ntext\' THEN 99 "
+		"WHEN \'numeric\' THEN 108 "
+		"WHEN \'nvarchar\' THEN 231 "
+		"WHEN \'real\' THEN 59 "
+		"WHEN \'smalldatetime\' THEN 58 "
+		"WHEN \'smallint\' THEN 52 "
+		"WHEN \'smallmoney\' THEN 122 "
+		"WHEN \'text\' THEN 35 "
+		"WHEN \'time\' THEN 41 "
+		"WHEN \'tinyint\' THEN 48 "
+		"WHEN \'uniqueidentifier\' THEN 36 "
+		"WHEN \'varbinary\' THEN 165 "
+		"WHEN \'varchar\' THEN 167 "
+		"WHEN \'xml\' THEN 241 "
+		"ELSE CASE "
+				"WHEN T2.name IS NOT NULL OR t.typbasetype = 0 THEN " 
```

Comment:
T2.name (sys.types.name) will never be NULL, why this check then?

Reply:
sys.types.name is obtained from translate_pg_type_to_tsql function. If invalid oid is provided, then the function will return NULL. To handle that case, this check is kept. Also, previously we had this check as well.

Reply:
Updated in latest rev